lib/auth/auth_with_roles.go

```diff
@@ -4416,10 +4416,13 @@ func (a *ServerWithRoles) GetSnowflakeSession(ctx context.Context, req types.Get
 	if session.GetSubKind() != types.KindSnowflakeSession {
 		return nil, trace.AccessDenied("GetSnowflakeSession only allows reading sessions with SubKind Snowflake")
 	}
-	// Users can only fetch their own app sessions.
-	if err := a.currentUserAction(session.GetUser()); err != nil {
-		if err := a.action(apidefaults.Namespace, types.KindDatabase, types.VerbRead); err != nil {
-			return nil, trace.Wrap(err)
+	// Check if this a database service.
+	if !a.hasBuiltinRole(types.RoleDatabase) {
+		// Users can only fetch their own web sessions.
+		if err := a.currentUserAction(session.GetUser()); err != nil {
```

Comment:
Here we used to check the session user. With this change, won't a user be able to read someone else's tokens?

Reply:
We allowed `GetSnowflakeSessions` (the plural form) with `VerbList`/`VerbRead` before this PR, and `GetAppSession`/`GetWebSession` both allow this kind of access. `VerbRead` includes secrets, `VerbReadNoSecrets` can be given to users who shouldn't be able to read secrets. Reading web sessions is not something that should be granted to normal users and the predefined access/auditor/editor roles do not have it.